src/Integration/ProjectToLanguageMapper.cs

```diff
@@ -47,34 +48,25 @@ public interface IProjectToLanguageMapper
     public class ProjectToLanguageMapper : IProjectToLanguageMapper
     {
         private readonly ICMakeProjectTypeIndicator cmakeProjectTypeIndicator;
+        private readonly IJsTsProjectTypeIndicator jsTsProjectTypeIndicator;
 
         internal static readonly IDictionary<Guid, Language> KnownProjectTypes = new Dictionary<Guid, Language>()
         {
-            { new Guid(ProjectSystemHelper.CSharpProjectKind), Language.CSharp },
-            { new Guid(ProjectSystemHelper.VbProjectKind), Language.VBNET },
-            { new Guid(ProjectSystemHelper.CSharpCoreProjectKind),  Language.CSharp },
-            { new Guid(ProjectSystemHelper.VbCoreProjectKind), Language.VBNET },
-            { new Guid(ProjectSystemHelper.CppProjectKind), Language.Cpp }
+            {new Guid(ProjectSystemHelper.CSharpProjectKind), Language.CSharp},
+            {new Guid(ProjectSystemHelper.VbProjectKind), Language.VBNET},
+            {new Guid(ProjectSystemHelper.CSharpCoreProjectKind), Language.CSharp},
+            {new Guid(ProjectSystemHelper.VbCoreProjectKind), Language.VBNET}
         };
 
         [ImportingConstructor]
-        public ProjectToLanguageMapper(ICMakeProjectTypeIndicator cmakeProjectTypeIndicator)
+        public ProjectToLanguageMapper(ICMakeProjectTypeIndicator cmakeProjectTypeIndicator,
+            IJsTsProjectTypeIndicator jsTsProjectTypeIndicator)
         {
             this.cmakeProjectTypeIndicator = cmakeProjectTypeIndicator;
+            this.jsTsProjectTypeIndicator = jsTsProjectTypeIndicator;
         }
 
-        /// <summary>
-        /// Returns the supported Sonar language for the specified project or Unknown
-        /// if no languages are supported
-        /// </summary>
-        /// <returns>
-        /// Previously the code assumed a one-to-one mapping between project types and languages.
-        /// The worked when the only supported languages were C# and VB. It doesn't work now that
-        /// connected mode is supported for C++ projects (which can have both C++ and C files).
-        /// New code should call <see cref="GetAllBindingLanguagesForProject(EnvDTE.Project)"/> instead
-        /// and handle the fact that there could be multiple supported languages.
-        /// </returns>
-        private Language GetLanguageForProject(Project dteProject)
+        public IEnumerable<Language> GetAllBindingLanguagesForProject(Project dteProject)
```

Comment:
This method needed a small clean up since a project can be both CSharp AND have js.